crates/router/src/compatibility/stripe/customers/types.rs

```diff
@@ -219,10 +219,11 @@ impl From<api::CustomerPaymentMethodsListResponse> for CustomerPaymentMethodList
     }
 }
 
+// Check this in review
 impl From<api_types::CustomerPaymentMethod> for PaymentMethodData {
     fn from(item: api_types::CustomerPaymentMethod) -> Self {
         Self {
-            id: item.payment_token,
+            id: item.payment_token.unwrap_or("".to_string()),
```

Comment:
We should ideally make this optional. Check with any stakeholders whether this would affect them negatively.

Reply:
Looks like no one's using this as of now, have made this optional.